### PR TITLE
[WIP] EIP roles

### PR DESCRIPTION
**Pull request by scothis**
Add support for [Enterprise Integration Pattern roles](https://docs.google.com/document/d/1nc9w2LM4K0fZoPyRwWZbRcgBno9QECMKABJTHVbP7-0/edit)

Builds on top of #4 for convince, but can be decoupled.

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -162,11 +160,57 @@ _Describes a linkage between a Subscribable and a Targetable and/or Sinkable._
 
 ### Life Cycle
 
+<<<<<<< HEAD
```
Comment (matzew) [inline docs/spec/spec.md]: conflict ? @scothis

File: docs/spec/spec.md [hunk 2]
```diff
@@ -33,15 +33,14 @@ cannot receive events._
 | ------------- | ---------------------------------- | ---------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------- |
 | provisioner\* | ProvisionerReference               | The provisioner used to create any backing resources and configuration.                                                | Immutable.                                                |
 | arguments     | runtime.RawExtension (JSON object) | Arguments passed to the provisioner for this specific source.                                                          | Arguments must validate against provisioner's parameters. |
-| channel       | ObjectRef                          | Specify an existing channel to use to emit events. If empty, create a new Channel using the cluster/namespace default. | Source will not emit events until channel exists.         |
+| channel\*     | ObjectRef                          | Specify a Channel to target.                                                                                           | Source will not emit events until channel exists.         |
```
Comment (n3wscott) [inline docs/spec/spec.md:36]: channel is optional

File: docs/spec/spec.md [hunk 3]
```diff
@@ -310,7 +348,32 @@ non-controlling OwnerReference on the EventType resources it knows about.
 
 ### ChannelSubscriberSpec
 
+<<<<<<< HEAD
```
Comment (n3wscott) [inline docs/spec/spec.md]: conflict here too

File: docs/spec/spec.md (hunk 2 above)
Comment (scothis) [reply docs/spec/spec.md:36]: #4 makes `channel` required

**Pull request closed without merging**